src/Entity/OgMembership.php

```diff
@@ -320,6 +320,22 @@ public function preSave(EntityStorageInterface $storage) {
       throw new \LogicException('OG membership can not be created for an empty or anonymous user.');
     }
 
+    if (!$this->get('entity_id')->value) {
+      // Group was not set.
+      throw new \LogicException('Membership cannot be set for an empty or an unsaved group.');
```

Comment:
Looking at \Drupal\Core\Entity\ContentEntityBase::preSave, I see that entities already fail to save if validationRequired is TRUE. I'm not sure how that property gets set to TRUE. Perhaps it is part of the entity annotation. I think thats we need to do. Then no special code is needed in presave().